scopus-import/src/main/java/no/sikt/nva/scopus/conversion/ContributorExtractor.java

```diff
@@ -137,40 +139,50 @@ private boolean isSamePerson(AuthorTp author, Contributor contributor) {
 
     private void generateContributorFromAuthorOrCollaboration(Object authorOrCollaboration, AuthorGroupTp authorGroupTp,
                                                               PersonalnameType correspondencePerson) {
-        if (authorOrCollaboration instanceof AuthorTp) {
-            generateContributor((AuthorTp) authorOrCollaboration, authorGroupTp, correspondencePerson);
+        if (authorOrCollaboration instanceof AuthorTp authorTp) {
+            generateContributor(authorTp, authorGroupTp, correspondencePerson);
         } else {
             generateContributorFromCollaborationTp((CollaborationTp) authorOrCollaboration, authorGroupTp,
                                                    correspondencePerson);
         }
     }
 
-    private void generateContributor(AuthorTp author, AuthorGroupTp authorGroup,
+    private void generateContributor(AuthorTp author,
+                                     AuthorGroupTp authorGroup,
                                      PersonalnameType correspondencePerson) {
-        var cristinPersonUri = piaConnection.getCristinPersonIdentifier(author.getAuid());
-        var cristinPerson = cristinConnection.getCristinPersonByCristinId(cristinPersonUri);
-        var cristinOrganizationUri = getCristinOrganizationUri(authorGroup);
-        var cristinOrganization = cristinConnection.getCristinOrganizationByCristinId(cristinOrganizationUri);
-        contributors.add(cristinPerson.map(
-                person -> CristinContributorExtractor.generateContributorFromCristin(person, author,
-                                                                                     correspondencePerson,
-                                                                                     cristinOrganization))
-                             .orElseGet(() -> generateContributorFromAuthorTp(author, authorGroup, correspondencePerson,
-                                                                              cristinOrganization)));
-    }
-
-    @Nullable
-    private URI getCristinOrganizationUri(AuthorGroupTp authorGroup) {
+
```

Comment:
The previous code was a bit confusing, hopefully this is clearer in its intention. @anetteOlli @truhacevkir check this with some diligence please as I tried, but may not have fully grasped the point.